### PR TITLE
Update Neyman threshold when changing runner_args

### PR DESCRIPTION
**Pull request by hammannr**
For shape parameters such as the WIMP mass, we haven't implemented anything that would allow resetting its nominal value after initialization. However, this is done to obtain the ``poi_rate_multiplier`` for a given ``poi_expectation``, which means that the poi_rate_multiplier values corresponding to a given threshold might be wrong in the end.
I did the following to fix this issue and prevent us from running into a similar issue in the future:
  - Initialize a runner for each runner_arg in the ``NeymanConstructor``. This takes a bit longer and may sometimes be overkill but it works.
  - Introduce a derived boolean property for a parameter called ``static`` (see #81 ), which is true if the parameter is
    - not fittable
    - neither a livetime nor rate (ptype)
    - has no blueice anchors
  - If one tries to override the nominal value of a static parameter or call parameters with a value for it that is different to its nominal value, alea will throw an error.

### CODE REVIEW

**Comment by hammannr** [comment]
I think it would be really neat if the static flag would set itself though I found that this is not really practical for our simple examples and thus I went back to making it a hand-settable flag.

**Comment by hammannr** [comment]
In principle one could also get around loading the runner explicitly by just obtaining the ``poi_expectation`` and ``poi_value`` from ``results[true_name]``, right? I didn't do that now because I wanted to change as little as possible but is there a reason why we should not do that @dachengx ?

File: alea/model.py [hunk 1]
```diff
@@ -95,16 +95,19 @@ def __init__(
         self._confidence_interval_kind = confidence_interval_kind
         self.confidence_interval_threshold = confidence_interval_threshold
         self.asymptotic_dof = asymptotic_dof
-        self._define_parameters(parameter_definition)
+        nominal_values = kwargs.get("nominal_values", {})
+        self._define_parameters(parameter_definition, nominal_values)
 
         self._check_ll_and_generate_data_signature()
-        self.set_nominal_values(**kwargs.get("nominal_values", {}))
```
Comment (dachengx) [inline alea/model.py:101]: Is the nearby functionality different from simply using `set_nominal_values`? @hammannr

Comment (hammannr) [reply alea/model.py:101]: I had to change it since otherwise the static parameters would complain if the nominal value were updated in this step

Comment (hammannr) [reply alea/model.py:101]: So the outcome is the same. It only had to be changed to accommodate the static parameter functionality

**Comment by dachengx** [comment]
> In principle one could also get around loading the runner explicitly by just obtaining the ``poi_expectation`` and ``poi_value`` from ``results[true_name]``, right? I didn't do that now because I wanted to change as little as possible but is there a reason why we should not do that @dachengx ?

Logically the thresholds json should be controlled by runner configuration yaml but not the files stored. But it is still beneficial to check the consistency between `result` and calculated rate multiplier? 

Better put this in another PR, as an enhancement.

File: alea/model.py (hunk 1 above)
Comment (dachengx) [reply alea/model.py:101]: Then maybe `set_nominal_values` is not checked by pytest. Would you add a test to increase the coverage?

Comment (hammannr) [reply alea/model.py:101]: Actually, now it's no longer in use so I would suggest to simply trash the function

**Comment by hammannr** [comment]
For some reason, pre-commit complains about `_uncertainty` in `parameters`, which I didn't touch:
```
alea/parameters.py:78: error: Cannot determine type of "_uncertainty"  [has-type]
alea/parameters.py:79: error: Cannot determine type of "_uncertainty"  [has-type]
alea/parameters.py:81: error: Cannot determine type of "_uncertainty"  [has-type]
```

**Comment by dachengx** [comment]
> For some reason, pre-commit complains about `_uncertainty` in `parameters`, which I didn't touch:
> 
> ```
> alea/parameters.py:78: error: Cannot determine type of "_uncertainty"  [has-type]
> alea/parameters.py:79: error: Cannot determine type of "_uncertainty"  [has-type]
> alea/parameters.py:81: error: Cannot determine type of "_uncertainty"  [has-type]
> ```

I will try to fix it.

**Review by dachengx — APPROVED** [review]

**Pull request merged**